### ISSUE TITLE
Routes for some 3rd-party resources not working

### ISSUE THREAD

**Issue by nasearle** [GoogleChrome/workbox#1447]
I have the following two routes in my service worker for Google fonts and MDL styles:

```
workbox.routing.registerRoute(
  new RegExp('https://fonts.(?:googleapis|gstatic).com/(.*)'),
  workbox.strategies.staleWhileRevalidate({
    cacheName: 'googleapis',
    plugins: [
      new workbox.expiration.Plugin({
        maxEntries: 30,
      }),
    ],
  }),
);

workbox.routing.registerRoute(
  new RegExp ('https://code.getmdl.io/(.*)'),
  workbox.strategies.staleWhileRevalidate({
    cacheName: 'furniture-store-mdl',
    plugins: [
      new workbox.expiration.Plugin({
        maxAgeSeconds: 7 * 24 * 60 * 60 // one week
      })
    ]
  })
```

Note that the first route is copied directly from [Common Recipes](https://developers.google.com/web/tools/workbox/guides/common-recipes), but the `cacheFirst` strategy is replaced with `staleWhileRevalidate`.

On the second page-load I can see that both routes are caching the responses, as expected:

![screen shot 2018-04-25 at 10 26 51 am](https://user-images.githubusercontent.com/14795039/39262092-44afad04-4873-11e8-82e0-47b0e70a32f0.png)

![screen shot 2018-04-25 at 10 28 04 am](https://user-images.githubusercontent.com/14795039/39262126-5df3631e-4873-11e8-93df-42329a89a3ee.png)

However, on subsequent loads I see that only the fonts are found in the cache and the MDL request goes on to the network:

![screen shot 2018-04-25 at 10 30 38 am](https://user-images.githubusercontent.com/14795039/39262359-02cb559a-4874-11e8-9aa1-9f870bc6a3f3.png)

![screen shot 2018-04-25 at 10 30 51 am](https://user-images.githubusercontent.com/14795039/39262367-05de9224-4874-11e8-9ddc-6fdff60fe5f4.png)

In the Network tab, I can see requests for Google fonts and MDL CSS are coming from the service worker and then again from workbox-core.dev.js:

![screen shot 2018-04-25 at 10 36 39 am](https://user-images.githubusercontent.com/14795039/39262747-1e5a0ea4-4875-11e8-88b8-4c6e5eebda9f.png)

When I take the app offline, neither the fonts nor the MDL CSS appear on the page. The log for the fonts is the same as before (response found in the cache), but fonts are not showing up on the page:

![screen shot 2018-04-25 at 10 30 38 am](https://user-images.githubusercontent.com/14795039/39262359-02cb559a-4874-11e8-9aa1-9f870bc6a3f3.png)

It looks like the MDL requests are not going through the `staleWhileRevalidate` route, and instead it logs:

![screen shot 2018-04-25 at 11 00 14 am](https://user-images.githubusercontent.com/14795039/39263722-dd67930a-4877-11e8-91d2-95fcdda83f80.png)

The result is an unstyled page with no fonts while offline. I tried using `networkFirst` and get the same issue.

To reproduce the issue:

```
git clone https://github.com/google-developer-training/pwa-ecommerce-demo.git
cd pwa-ecommerce-demo/solution
git checkout v2-ecommerce-app
npm install
npm run serve
```

Open browser to localhost:8080

**Comment by jeffposnick** [comment]
I believe you're running into the issue resolved in https://github.com/GoogleChrome/workbox/pull/1422, which was included in the Workbox v3.1.0 release.

When I checked out your code locally and ran it with v3.1.0, things worked as expected.

(Your service worker has the v3.0.0 CDN URL hardcoded, so you were still using that, despite having updated the dependency in `package.json`.)

**Comment by nasearle** [comment]
Ah ok great! Thanks Jeff
